### PR TITLE
Webdriver

### PR DESCRIPTION
**Pull request by AdaRoseCannon**
Adding integration tests

### CODE REVIEW

File: wdio.conf.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	// time. Depending on the number of capabilities, WebdriverIO launches several test
+	// sessions. Within your capabilities you can overwrite the spec and exclude option in
+	// order to group specific specs to a specific capability.
+	//
+	// If you have trouble getting all important capabilities together, check out the
+	// Sauce Labs platform configurator - a great tool to configure your capabilities:
+	// https://docs.saucelabs.com/reference/platforms-configurator
+	//
+	capabilities: [{
+		browserName: 'firefox'
```
Comment (JakeChampion) [inline wdio.conf.js]: Could this be chrome instead of firefox?

File: wdio.conf.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	// ==================
+	// Specify Test Files
+	// ==================
+	// Define which test specs should run. The pattern is relative to the directory
+	// from which `wdio` was called. Notice that, if you are calling `wdio` from an
+	// NPM script (see https://docs.npmjs.com/cli/run-script) then the current working
+	// directory is where your package.json resides, so `wdio` will be called from there.
+	//
+	specs: [
+		'./tests/specs/**/*.js'
```
Comment (JakeChampion) [inline wdio.conf.js]: Could we have the integration tests in their own folder (named `integration`)?

File: wdio.conf.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+const denodeify = require('denodeify');
+const selenium = require('selenium-standalone');
+const installSelenium = denodeify(selenium.install.bind(selenium));
+const startSeleniumServer = denodeify(selenium.start.bind(selenium));
+
+/*
+ * Installs Selenium and starts the server, ready to control browsers
+*/
+
+function installAndStartSelenium () {
```
Comment (JakeChampion) [inline wdio.conf.js]: :ok_hand:

File: server/middleware/auth.js [hunk 4]
```diff
@@ -1,7 +1,14 @@
+'use strict';
```
Comment (JakeChampion) [inline server/middleware/auth.js]: I don't believe modules need to explicitly declare `'use strict';`

File: wdio.conf.js (hunk 2 above)
Comment (AdaRoseCannon) [inline wdio.conf.js]: sure

File: wdio.conf.js (hunk 1 above)
Comment (AdaRoseCannon) [inline wdio.conf.js]: sure

File: server/middleware/auth.js (hunk 4 above)
Comment (AdaRoseEdwards) [inline server/middleware/auth.js]: Have the linting settings changed? My text editor warns me.

**Comment by AdaRoseCannon** [comment]
It was because it had not finished parsing the body by the time the script was run. Which stopped the viewer from running occasionally. Seemed to be flaky and only when running through the testing suite though, could not reproduce by just running the server and looking in the browser.

**Pull request merged**